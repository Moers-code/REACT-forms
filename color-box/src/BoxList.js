import React, { useState } from "react";
import Box from "./Box";
import { v4 as uuidv4 } from "uuid";
import BoxForm from "./BoxForm";

const BoxList = () => {
	const [boxes, setBoxes] = useState([]);
	const addBox = (newBox) => {
		setBoxes((boxes) => [...boxes, { id: uuidv4(), ...newBox }]);
	};

	const remove = (id) =>
		setBoxes((boxes) => boxes.filter((box) => box.id !== id));

	return (
		<div>
			<BoxForm addBox={addBox} />
			<div>
				{boxes.map((box) => (
					<Box
						key={box.id}
						id={box.id}
						remove={remove}
						width={parseInt(box.width)}
						height={parseInt(box.height)}
						backgroundColor={box.backgroundColor}
					/>
				))}
			</div>
		</div>
	);
};

export default BoxList;
