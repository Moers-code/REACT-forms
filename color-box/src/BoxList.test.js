import React from "react";
import {
	render,
	fireEvent,
	getByText,
	cleanup,
	waitFor,
} from "@testing-library/react";
import BoxList from "./BoxList";

afterEach(cleanup);

it("should render without crashing", () => {
	render(<BoxList />);
});

it("should match snapshot", () => {
	const { asFragment } = render(<BoxList />);
	expect(asFragment()).toMatchSnapshot();
});

it("can add a new box", async () => {
	const mockAddBox = jest.fn();
	const { getByLabelText, getByText } = render(<BoxList />);
	const backgroundColorInput = getByLabelText("Color");
	const widthInput = getByLabelText("Width");
	const heightInput = getByLabelText("Height");
	const submitButton = getByText("Add Box");

	fireEvent.change(backgroundColorInput, { target: { value: "black" } });
	fireEvent.change(widthInput, { target: { value: "50" } });
	fireEvent.change(heightInput, { target: { value: "50" } });
	fireEvent.click(submitButton);

	await waitFor(() => {
		const newBox = getByText("x");
		expect(newBox).toBeInTheDocument();
	});
});
