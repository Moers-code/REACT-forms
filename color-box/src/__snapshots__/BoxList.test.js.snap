// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="backgroundColor"
      >
        Color
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        placeholder="Background Color"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width
      </label>
      <input
        id="width"
        name="width"
        placeholder="Width"
        type="number"
        value=""
      />
      <label
        for="height"
      >
        Height
      </label>
      <input
        id="height"
        name="height"
        placeholder="Height"
        type="number"
        value=""
      />
      <button>
        Add Box
      </button>
    </form>
    <div />
  </div>
</DocumentFragment>
`;
